### PR TITLE
Implement ItemSizingStrategy.MeasureFirstItem on Android

### PR DESCRIPTION
**Pull request by hartez**
### Issues Resolved ### 

- implements part of #3172

### API Changes ###
 
 None

### Platforms Affected ### 

- Android

### Behavioral/Visual Changes ###

None

### Before/After Screenshots ### 

Not applicable

### Testing Procedure ###

Control Gallery, navigate to CollectionView Gallery ->Item Size Galleries ->ItemSizing Strategy

The initial selection is MeasureFirstItem; all of the items should have the same size. If you switch to MeasureAllItems, the items will have varying sizes.

### PR Checklist ###

- [ ] Has automated tests 
- [x] Rebased on top of the target branch at time of PR
- [x] Changes adhere to coding standard

### CODE REVIEW

File: Xamarin.Forms.Platform.Android/CollectionView/ItemsViewAdapter.cs [hunk 1]
```diff
@@ -65,10 +79,21 @@ public override RecyclerView.ViewHolder OnCreateViewHolder(ViewGroup parent, int
...
+		ItemContentView CreateItemContentView(Context context)
+		{
+			if(ItemsView.ItemSizingStrategy == ItemSizingStrategy.MeasureAllItems
+				|| _size == null)
+			{
+				return new ItemContentView(context);
+			}
+
+			return new ItemContentView(context);
+		}
```
Comment (paymicro) [inline Xamarin.Forms.Platform.Android/CollectionView/ItemsViewAdapter.cs]: What is the meaning of this method if it always returns `new ItemContentView(context)`?

Comment (hartez) [reply Xamarin.Forms.Platform.Android/CollectionView/ItemsViewAdapter.cs]: Leftover from an earlier implementation. I'll remove it.

**Comment by hartez** [comment]
> Also why does the strategy MeasureAllItems ignore the first 2 elements?

The strategy doesn't ignore them, the test does. The template for it has an IValueConverter which forces the first few items to have a different width/height than the rest. That way, we can see a difference between measuring the first item and using that size, or measuring all the items individually.

**Comment by hartez** [comment]
> After changing the ItemSizingStrategy to MeasureFirstItem, the images inside have a wrong vertical offset.

Huh. That is _weird_. I'll take a look at that on Monday.

**Comment by paymicro** [comment]
> Huh. That is _weird_. I'll take a look at that on Monday.

I think that this is due to a change in the position of the entire CollectionView. As the text above becomes 2 lines.

**Comment by hartez** [comment]
> > Huh. That is _weird_. I'll take a look at that on Monday.
> 
> I think that this is due to a change in the position of the entire CollectionView. As the text above becomes 2 lines.

I think you're right. 

Well, I don't think it's reason enough to hold this back from getting merged. We can always log that as a separate bug in the future if we need to.

**Pull request merged**